ykman/cli/piv.py

```diff
@@ -792,6 +792,68 @@ def unblock_pin(ctx, puk, new_pin):
             show_default=False, hide_input=True, err=True)
     controller.unblock_pin(puk, new_pin)
 
+@piv.command('read-object')
+@click_pin_option
+@click.pass_context
+@click.option(
+        '-i', '--id', 'object_id',
+        callback=lambda ctx, param, value: int(value, 16),
+        metavar='HEX', help='Id of PIV object.', required=True)
+def read_object(ctx, pin, object_id):
+    """
+    Read arbitrary PIV object.
+
+    Read PIV object by providing the object id.
+    """
+
+    controller = ctx.obj['controller']
+
+    def do_read_object(retry=True):
+        try:
+            click.echo(controller.get_data(object_id))
+        except APDUError as e:
+            if e.sw == SW.NOT_FOUND:
+                ctx.fail('No data found.')
+            elif e.sw == SW.SECURITY_CONDITION_NOT_SATISFIED:
+                _verify_pin(ctx, controller, pin)
+                do_read_object(retry=False)
+            else:
+                raise
+
+    do_read_object()
+
+@piv.command('write-object')
+@click_pin_option
+@click_management_key_option
+@click.pass_context
+@click.option(
+        '-i', '--id', 'object_id',
+        callback=lambda ctx, param, value: int(value, 16),
+        metavar='HEX', help='Id of PIV object.', required=True)
+@click.argument('data', type=click.File('rb'), metavar='DATA')
```

Comment:
It isn't clear from the help text that this should be a file name, or that `-` resolves to standard input.